hera_cal/abscal.py

```diff
@@ -981,510 +2636,319 @@ def TT_Phi_gain_arr(self):
-                    merged_gains.append(AC.TT_Phi_gain)
-                # test for convergence
-                if len(merged_gains) > 2:
-                    gains_before = merge_gains(merged_gains[:-2])
-                    gains_after = merge_gains(merged_gains)
-                    # take L2 norm over antennas and times
-                    gains_norm = np.linalg.norm([gains_after[k] for k in gains_after.keys()], axis=(0, 1))
-                    delta_gains_norm = np.linalg.norm([gains_after[k] - gains_before[k] for k in gains_after.keys()], axis=(0, 1))
-                    # take median over frequency to avoid the effect of band edges and RFI
-                    echo("TT_phs_cal convergence criterion: " + str(np.median(delta_gains_norm / gains_norm)), verbose=verbose)
-                    if np.median(delta_gains_norm / gains_norm) < phs_conv_crit:
-                        break
-                merged_gains = [merge_gains(merged_gains)]
-
-        if gen_amp_cal:
-            if all_antenna_gains:
-                raise ValueError("can't run gen_amp_cal when all_antenna_gains is True")
-            AC.amp_logcal(verbose=verbose)
-            cal_flags = odict(list(map(lambda k: (k, np.zeros_like(AC.ant_eta_gain[k], np.bool)), AC.ant_eta_gain.keys())))
-            apply_cal.calibrate_in_place(AC.data, AC.ant_eta_gain, AC.wgts, cal_flags, gain_convention='divide', flags_are_wgts=True)
-            merged_gains.append(AC.ant_eta_gain)
-            merged_gains = [merge_gains(merged_gains)]
-
-        if gen_phs_cal:
-            if not (delay_cal or delay_slope_cal):
-                echo("it is recommended to run a delay_cal or delay_slope_cal before gen_phs_cal", verbose=verbose)
-            if all_antenna_gains:
-                raise ValueError("can't run gen_phs_cal when all_antenna_gains is True")
-            AC.phs_logcal(verbose=verbose)
-            cal_flags = odict(list(map(lambda k: (k, np.zeros_like(AC.ant_phi_gain[k], np.bool)), AC.ant_phi_gain.keys())))
-            apply_cal.calibrate_in_place(AC.data, AC.ant_phi_gain, AC.wgts, cal_flags, gain_convention='divide', flags_are_wgts=True)
-            merged_gains.append(AC.ant_phi_gain)
-            merged_gains = [merge_gains(merged_gains)]
-
-        # collate gains
-        if len(merged_gains) == 0:
-            raise ValueError("abscal_run executed without any calibration arguments set to True")
-        gain_dict = merge_gains(merged_gains)
-        flag_dict = odict(list(map(lambda k: (k, np.zeros((Ntimes, Nfreqs), np.bool)), gain_dict.keys())))
-        gain_pols = AC.gain_pols
-        gain_keys = gain_dict.keys()
-
-        # join gain_flag_dict with flag_dict
-        for k in flag_dict:
-            if k in gain_flag_dict:
-                flag_dict[k] += gain_flag_dict[k]
-
-    # make blank gains if no modelfiles
+def get_all_times_and_lsts(hd, solar_horizon=90.0, unwrap=True):
+    '''Extract all times and lsts from a HERAData object
```

Comment:
this only works for a `HERAData` instantiated with a `uvh5`, see `io.get_file_lsts` for how to do this for other file formats. Let's just have one function in the codebase to do this functionality, if you want to add `times` in addition to `lsts`, great!